src/net/sourceforge/kolmafia/persistence/ConcoctionDatabase.java

```diff
@@ -1252,6 +1253,58 @@ public static final void refreshConcoctions(boolean force) {
     ConcoctionDatabase.refreshConcoctionsNow();
   }
 
+  /**
+   * Get first index of usableList that has sortOrder non-NONE, assuming the list is sorted by
+   * sortOrder.
+   */
+  private static OptionalInt getFirstIndexNotNone() {
+    int low = 0;
+    int high = ConcoctionDatabase.usableList.size();
+    while (low < high) {
+      int mid = (low + high) / 2;
+      if (ConcoctionDatabase.usableList.get(mid).sortOrder == Concoction.Priority.NONE) {
+        low = mid + 1;
+      } else {
+        high = mid;
+      }
+    }
+    if (ConcoctionDatabase.usableList.get(low).sortOrder == Concoction.Priority.NONE) {
+      return OptionalInt.empty();
```

Comment:
Thanks. I don't think there will be a performance difference since you're doing a check either way, and this way absolutely ensures that if this returns a present int that the item at that index is not NONE.

Reply:
You're doing a check either way, but by doing the check first you don't have to do the binary search bit.

The only way that the chosen entry can have sort order NONE is if all items in the list have sort order NONE (as they are sorted), and so checking only the last entry means you can shift the check to the start.